### PR TITLE
Add 8 new IdP types

### PR DESCRIPTION
**Pull request by jackward-okta**
Amazon, GitHub, GitLab, Yahoo, Yahoo Japan, LINE, PayPal, Salesforce, and Amazon.

Will add these to the README when we go EA with them.

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Added unit tests?
- [ ] Added e2e tests
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Screenshot/Video:

image

### Reviewers:

- App Plat. team
- Sign-in Widget Owners

### Issue:

- [OKTA-424516](https://oktainc.atlassian.net/browse/OKTA-424516)

### CODE REVIEW

**Comment by pradeepdewda-okta** [comment]
@jackward-okta is this also for OIE - widget v2 ?

**Review by pradeepdewda-okta — APPROVED** [review]

**Comment by jackward-okta** [comment]
> @jackward-okta is this also for OIE - widget v2 ?

@pradeepdewda-okta helped me out over DMs about this—turns out it does in fact support v2 :)

**Review by colekurashige-okta — COMMENTED** [review]
It's missing the paypal sandbox IdP.

From our discussions on formatting, the name should be `PAYPAL_SANDBOX`.

By the way, should this be merged before the `okta-core` PR that changes how the type is sent/received? My instincts say "no."

And finally, I wonder if it would be possible to gate this behind the `METADATA_DRIVEN_SOCIAL_IDP` FF. I expect the answer is "no," which should be fine since there shouldn't be a way to integrate these IdPs otherwise, but if we could gate it that might be for the better. Granted, it does make removing the FF more annoying.

**Comment by colekurashige-okta** [comment]
> It's missing the paypal sandbox IdP.
> 
> From our discussions on formatting, the name should be `PAYPAL_SANDBOX`.
> 
> By the way, should this be merged before the `okta-core` PR that changes how the type is sent/received? My instincts say "no."
> 
> And finally, I wonder if it would be possible to gate this behind the `METADATA_DRIVEN_SOCIAL_IDP` FF. I expect the answer is "no," which should be fine since there shouldn't be a way to integrate these IdPs otherwise, but if we could gate it that might be for the better. Granted, it does make removing the FF more annoying.

I'm not requesting changes since I might be out tomorrow (just in case you want to merge), but I'd like you to hold off on merging before you have answers to these points/questions.

**Comment by jackward-okta** [comment]
> It's missing the paypal sandbox IdP. From our discussions on formatting, the name should be `PAYPAL_SANDBOX`.

Ah true, got that mixed up. Will re-name the `PAYPAL` one.

> By the way, should this be merged before the `okta-core` PR that changes how the type is sent/received? My instincts say "no."

I was thinking I'd add it in an undocumented way for now, and add them to the README when we go EA—is the worry that we'd change approaches when we're actually in it?

Of the three merge times (now, after the `type` change reaches prod, and after flip to EA) I don't know how much benefit we get from the second over the first, and the third actually makes some sense to me. Maybe we can just let this sit approved and coded until EA.

> And finally, I wonder if it would be possible to gate this behind the `METADATA_DRIVEN_SOCIAL_IDP` FF. I expect the answer is "no," which should be fine since there shouldn't be a way to integrate these IdPs otherwise, but if we could gate it that might be for the better. Granted, it does make removing the FF more annoying.

Yeah, your instinct is right. The self-hosted sign-in widget doesn't make any calls to the monolith to fetch feature flags or anything else, so we can't depend on them.

File: assets/sass/modules/_btns.scss
Comment (luishernandezcruz-okta) [inline assets/sass/modules/_btns.scss:231]: curious, why the `12px/24px` vs `10px` used in `social-auth-button-template`

**Review by luishernandezcruz-okta — APPROVED** [review]
LGTM 👍

**Review by mohamedelsherif-okta — APPROVED** [review]
Awesome 🚢

File: assets/sass/modules/_btns.scss
Comment (jackward-okta) [reply assets/sass/modules/_btns.scss:231]: There are two current approaches to the icon positioning and sizing:

- Facebook, LinkedIn, and Microsoft display 30x30px icons at 1x size shifted over 10px (the `10px` you mention)
- Apple and Google display 20x20px icons at 1x size shifted over 15px (overriding the `10px` you mention)

For the new icons, we wanted to display them at 0.5x size because it looks better on a retina display. We picked 24px wide [somewhat arbitrarily](https://oktainc.atlassian.net/browse/OKTA-422990), and packaged 48px icons. The resulting approach is:

- Our new IdPs display 48x48px icons at 0.5x size (the `24px` in your message) shifted over 12px (the `12px` in your message)

**Comment by jackward-okta** [comment]
Got word from Cole over Slack DM to add the `PAYPAL` IdP back in addition to `PAYPAL_SANDBOX`. Adding now

**Review by colekurashige-okta — APPROVED** [review]
LGTM pending PayPal changes

File: test/unit/helpers/dom/PrimaryAuthForm.js
Comment (jmelberg-okta) [inline test/unit/helpers/dom/PrimaryAuthForm.js:255]: Are these types unused purposefully?

Let's either add tests for them or remove them from our test helpers to avoid unused functions.

Comment (jackward-okta) [reply test/unit/helpers/dom/PrimaryAuthForm.js]: Not used—was just following along with the PR to add the `APPLE` IdP. Removed in d043e00

**Comment by jackward-okta** [comment]
Whoops, clicked the "Update Branch" button down here thinking it might update the codeowners status but nope it made a merge commit. Will squash it out later :)

**Comment by jackward-okta** [comment]
Karma tests were modified since I split my branch off! Amending this patch to fix it:

```diff
diff --git a/test/unit/spec/PrimaryAuth_spec.js b/test/unit/spec/PrimaryAuth_spec.js
index 2ae12a80..4db6cfe6 100644
--- a/test/unit/spec/PrimaryAuth_spec.js
+++ b/test/unit/spec/PrimaryAuth_spec.js
@@ -2699,7 +2699,7 @@ Expect.describe('PrimaryAuth', function() {
         const buttons = test.form.socialAuthButtons();
         expect(buttons.length).toBe(idpTypes.length);
         for (const [i, type] of idpTypes.entries()) {
-          expect(buttons.eq(i)).toHaveClass(`social-auth-${type.toLowerCase()}-button`);
+          expect(buttons.eq(i).attr('class')).toContain(`social-auth-${type.toLowerCase()}-button`);
         }
       });
     });
```

File: packages/@okta/i18n/src/properties/login.properties
Comment (jmelberg-okta) [inline packages/@okta/i18n/src/properties/login.properties:651]: Since this extends support for many more IdPs, can we update our [Social Login section](https://github.com/okta/okta-signin-widget/tree/2babf07e346349706dbc91d8ff9011e54db4e807#openid-connect) of the README to either link to or explicitly call out all supported IdPs?

**Review by jmelberg-okta — APPROVED** [review]
🌐  i18n changes look good!

Before merging, please:
- Confirm the base branch is correct. Usually we introduce changes like this as a part of weekly, but since this is EOM I can see why `master` is preferred.
- Update the commit message for follow the [Contributing Guidelines](https://github.com/okta/okta-signin-widget/blob/2babf07e346349706dbc91d8ff9011e54db4e807/CONTRIBUTING.md#commit-message-guidelines)
- List out newly supported IdPs in the README.

**Review by jackward-okta — COMMENTED** [review]
Thank you for your review @jmelberg-okta!

> Confirm the base branch is correct. Usually we introduce changes like this as a part of weekly, but since this is EOM I can see why master is preferred.

This sounds like a part of this repo's development process I'm not familiar with—do you build releases on certain branches? It looks like 5.11.0 is the latest release, but the latest branch is 5.10. Happy to change the base to wherever is preferred.

> Update the commit message for follow the Contributing Guidelines

Fixed!

**Comment by pradeepdewda-okta** [comment]
> Thank you for your review @jmelberg-okta!
> 
> > Confirm the base branch is correct. Usually we introduce changes like this as a part of weekly, but since this is EOM I can see why master is preferred.
> 
> This sounds like a part of this repo's development process I'm not familiar with—do you build releases on certain branches? It looks like 5.11.0 is the latest release, but the latest branch is 5.10. Happy to change the base to wherever is preferred.
> 
> > Update the commit message for follow the Contributing Guidelines
> 
> Fixed!

@jackward-okta Is there any urgency to get this released? If you are okay to publish this during a monthly release, then keeping this in `master` should be fine. Monthly release would be end of this month.

**Comment by jackward-okta** [comment]
> @jackward-okta Is there any urgency to get this released? If you are okay to publish this during a monthly release, then keeping this in master should be fine. Monthly release would be end of this month.

No urgency; monthly is fine by me. I'll leave it on this branch then.

**Pull request closed without merging**
